frontend/web/components/release-pipelines/StageStatus.tsx

```diff
@@ -0,0 +1,115 @@
+import ProjectStore from 'common/stores/project-store'
+import {
+  Environment,
+  Features,
+  StageAction,
+  StageTrigger,
+  StageTriggerType,
+} from 'common/types/responses'
+import classNames from 'classnames'
+import Icon from 'components/Icon'
+import { renderToString } from 'react-dom/server'
+
+import StageSummaryData from './StageSummaryData'
+
+export interface StageStatusProps {
+  stageOrder: number
+  stageName: string
+  totalStages: number
+  stageActions?: StageAction[]
+  stageTrigger?: StageTrigger
+  stageEnvironment?: number
+  featureInStage?: Features[number]
+  isCompleted?: boolean
+}
+
+const StageStatus = ({
+  featureInStage,
+  isCompleted,
+  stageActions,
+  stageEnvironment,
+  stageName,
+  stageOrder,
+  stageTrigger,
+  totalStages,
+}: StageStatusProps) => {
+  const isFeatureInStage = !!featureInStage
+  const showLine = totalStages > 1
+  const lastStage = stageOrder === totalStages - 1
+  const showLeftLine = showLine && stageOrder > 0
+  const showRightLine = showLine && stageOrder !== totalStages - 1
+
+  const isWaiting =
+    stageTrigger?.trigger_type === StageTriggerType.WAIT_FOR && isFeatureInStage
+  const isLastStage = stageOrder === totalStages - 1
+
+  const stageStyle = {
+    marginRight: lastStage ? '0px' : '24px',
+  }
+
+  const env = ProjectStore.getEnvironmentById(
+    stageEnvironment,
+  ) as unknown as Environment
+  const envName = env?.name
+
+  return (
+    <div
+      className='flex align-items-center gap-2 position-relative flex-1'
+      style={stageStyle}
+    >
+      {showLeftLine && (
+        <div
+          className={classNames('position-absolute line-left', {
+            'completed': isFeatureInStage,
```

Comment:
<img width="361" height="158" alt="image" src="https://github.com/user-attachments/assets/ee77182c-2f36-4c5c-91a4-d61f3aef3f92" />

Maybe we can use something like?
```
stage.order <= (stageHasFeature?.order ?? -1)
```